sklearn/ensemble/gradient_boosting.py

```diff
@@ -634,7 +646,8 @@ def _update_terminal_region(self, tree, terminal_regions, leaf, X, y,
         numerator = np.sum(y_ * sample_weight * np.exp(-y_ * pred))
         denominator = np.sum(sample_weight * np.exp(-y_ * pred))
 
-        if denominator == 0.0:
+        # prevents overflow and division by zero
+        if abs(denominator) < 1e-150:
```

Comment:
Thing is if your system is 32 bit denominator which is the result of `np.sum` can only be as low as (`np.finfo(np.float<arch>).tiny`) IIUC...